src/middlewares/upload_logo.ts

```diff
@@ -0,0 +1,111 @@
+  ResponseErrorForbiddenNotAuthorized,
+  ResponseErrorInternal,
+  ResponseErrorNotFound,
+  ResponseSuccessRedirectToResource
+} from "italia-ts-commons/lib/responses";
+import { NonEmptyString } from "italia-ts-commons/lib/strings";
+import {
+  getApimUser,
+  IExtendedUserContract,
+  isAdminUser
+} from "../apim_operations";
+import { AdUser } from "../bearer_strategy";
+import * as config from "../config";
+
+import { Logo as ApiLogo } from "../../generated/api/Logo";
+
+import {
+  fromLeft,
+  fromPredicate,
+  taskEither,
+  TaskEither,
+  tryCatch
+} from "fp-ts/lib/TaskEither";
+
+import { ErrorResponses, notificationApiClient } from "../controllers/services";
+
+import { toError } from "fp-ts/lib/Either";
+
+export const getApimUserTask = (
+  apiClient: ApiManagementClient,
+  authenticatedUser: AdUser
+): TaskEither<ErrorResponses, IExtendedUserContract> =>
+  tryCatch(
+    () => getApimUser(apiClient, authenticatedUser.emails[0]),
+    errors => ResponseErrorInternal(toError(errors).message)
+  ).foldTaskEither<ErrorResponses, IExtendedUserContract>(
+    error => fromLeft(error),
+    response =>
+      response.isSome()
+        ? taskEither.of(response.value)
+        : fromLeft(
+            ResponseErrorNotFound(
+              "API user not found",
+              "Cannot find a user in the API management with the provided email address"
+            )
+          )
+  );
+
+export const checkAdminTask = (
+  userAdmin: IExtendedUserContract
+): TaskEither<ErrorResponses, IExtendedUserContract> =>
+  fromPredicate(
+    (user: IExtendedUserContract) => isAdminUser(user),
+    _ => ResponseErrorForbiddenNotAuthorized
+  )(userAdmin);
+
+export const uploadServiceLogoTask = (
+  serviceId: NonEmptyString,
+  serviceLogo: ApiLogo
+): TaskEither<ErrorResponses, IResponseSuccessRedirectToResource<{}, {}>> =>
+  tryCatch(
+    () =>
+      notificationApiClient.uploadServiceLogo({
+        logo: serviceLogo,
+        serviceId
+      }),
+    errors => ResponseErrorInternal(toError(errors).message)
+  ).foldTaskEither(
+    error => fromLeft(error),
+    errorOrResponse =>
+      errorOrResponse && errorOrResponse.status === 201
+        ? taskEither.of(
+            ResponseSuccessRedirectToResource(
+              {},
+              `${config.logoUrl}/services/${serviceId}.png`,
+              {}
+            )
+          )
+        : fromLeft(ResponseErrorInternal(toError(errorOrResponse).message))
+  );
+
+export const uploadOrganizationLogoTask = (
+  organizationfiscalcode: NonEmptyString,
+  serviceLogo: ApiLogo
+): TaskEither<ErrorResponses, IResponseSuccessRedirectToResource<{}, {}>> =>
+  tryCatch(
+    () =>
+      notificationApiClient.uploadOrganizationLogo({
+        logo: serviceLogo,
+        organizationfiscalcode
+      }),
+    errors => ResponseErrorInternal(toError(errors).message)
+  ).foldTaskEither(
+    err => fromLeft(err),
+    errorOrResponse =>
+      errorOrResponse && errorOrResponse.status === 201
```

Comment:
ditto

Reply:
Yes, (as for `uploadServiceLogo`) can be undefined according to `uploadOrganizationLogo` defined in `api_client.ts`.

Reply:
ditto